pandas/core/internals/blocks.py

```diff
@@ -523,6 +525,8 @@ def replace(
         inplace: bool = False,
         # mask may be pre-computed if we're called from replace_list
         mask: npt.NDArray[np.bool_] | None = None,
+        original_blocks: list[Block] = [],
```

Comment:
Do we need to take a list here, since we're only setting the ref to 1 block?

Reply:
I want to use the same as in https://github.com/pandas-dev/pandas/pull/50428, but want to get this merged before